lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php

```diff
@@ -337,8 +337,8 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
                 $metadata->setSequenceGeneratorDefinition(
                     [
                         'sequenceName' => (string) $seqGenerator['sequence-name'],
-                        'allocationSize' => (string) $seqGenerator['allocation-size'],
-                        'initialValue' => (string) $seqGenerator['initial-value']
+                        'allocationSize' => (int) $seqGenerator['allocation-size'],
+                        'initialValue' => (int) $seqGenerator['initial-value']
```

Comment:
Tbh, sequences are not necessarily int-based

Reply:
They can be floats? I do not think I can do `CREATE SEQUENCE INCREMENT ` with anything but integer. Neither PostgreSQL, nor oracle https://docs.oracle.com/database/121/SQLRF/statements_6017.htm#SQLRF01314

Reply:
Postgres only supports bigint:
> Sequences are based on bigint arithmetic, so the range cannot exceed the range of an eight-byte integer (-9223372036854775808 to 9223372036854775807).
> https://www.postgresql.org/docs/9.5/static/sql-createsequence.html

But on 32-bit platform, bigint would be string/float...

Reply:
Interesting. That would be still on PostgreSQL side though. The bug at highlighted line in referenced issue is in PHP side, where SequenceGenerator trying to predict maximum value. It is expecting PHP integer, as enforced by the cast, to do the addition for the prediction. But if value comes from XML parser, prediction calculation fails having second addend being string.

Reply:
Or are you saying because of this 32-bit corner case, the schema mapping should not be touched, but the cast should be fixed instead wherever the parameters are used? In this case, are you saying xml parser is not root cause, that `SequenceGenerator` should be fixed?

Reply:
Also turns out migrations wants the value to be string. If driver casts to integer itself, migration diff generates INCREMENT BY 0. Unit tests here do not notice that potential regression. so I guess the crash is better fixed at the SequenceGenerator level.

Reply:
sorry, discovered actual root cause. Thanks @Majkl578 for making me think further about it. Turns out xml driver is not supplying default values for optional fields when they are omitted in mapping file. This results in empty $definition keys, which causes "non numeric value" error during addition, *not* because it is a casting problem as I originally thought. 

Will redo the fix by either validating input in `ClassMetadataInfo::setSequenceGeneratorDefinition`  where it already does a little bit of parsing validation for sequence_name, but not the other parameters, or maybe I can find why XSD schema is not applied by XML driver.